### ISSUE TITLE
Use same JPEG 'Quality' when saving image

### ISSUE THREAD

**Issue by DeanIsMe** [sylikc/jpegview#82]
_More of a suggestion that an issue_
If I use JPEGView open a JPEG with quality=50 then save it, the new file has quality=85.
The file size grows from 727 KB to 1130 KB.
Of course, no additional quality is added to the image so this process just makes the image take up more storage space.

A more appropriate default behaviour would be: **when saving JPEG, use the same quality setting as the original file.**

This will also preserve quality in JPEGs that use low compression. E.g. opening a JPEG with quality=95, it would be best to save as quality=95 to avoid losing detail.

**Comment by Pentaphon** [comment]
The default setting is 85 using the settings.ini so you have to change it manually for now.

**Comment by DeanIsMe** [comment]
Thanks @pureby, that's really helpful to know.
I guess it's a lot harder to decide on the best JPEG quality. More advanced decision making could consider the original image's quality and the user's preference, but that's of course more effect to figure out and implement.

**Comment by Pentaphon** [comment]
I think it should always be at 100 since it's just an image viewer. Let an image editor change the quality if need be.

**Comment by sylikc** [comment]
I actually had looked up "how to know what the previous save quality was"... and apparently there's no clear answer.  You can guess what it was saved before, but you never really know

**Comment by Pentaphon** [comment]
> I actually had looked up "how to know what the previous save quality was"... and apparently there's no clear answer. You can guess what it was saved before, but you never really know

Sounds like more reason to make an jpegview default to 100 so that the user doesn't destroy their images when simply converting an image to another format without realizing that they are defaulted to 85.

**Comment by sylikc** [comment]
> Sounds like more reason to make jpegview default to 100 so that the user doesn't destroy their images when simply converting an image to another format without realizing that they are defaulted to 85.

It's actually not a really true statement... saving JPEG is lossy no matter whether it's 100 or not.  Setting it to 100 would actually give users a false sense of security.  I've worked with JPEGs extensively, and if you search around the web you'll see that this is the case.  Size of an image may actually increase, but no new image data is added, and you've actually lost data.

I'll note this feature request in the pinned issue, but it is not likely I'd ever set it to 100 by default.

[So you think saving a JPEG at quality 97 will prevent generation loss?](https://www.reddit.com/r/photography/comments/4f1pfp/so_you_think_saving_a_jpeg_at_quality_97_will/)
[[Is JPEG lossless when quality is set to 100?](https://stackoverflow.com/questions/7982409/is-jpeg-lossless-when-quality-is-set-to-100)](https://stackoverflow.com/questions/7982409/is-jpeg-lossless-when-quality-is-set-to-100/48578831#48578831)

**Comment by Pentaphon** [comment]
> It's actually not a really true statement... saving JPEG is lossy no matter whether it's 100 or not. Setting it to 100 would actually give users a false sense of security. I've worked with JPEGs extensively, and if you search around the web you'll see that this is the case. Size of an image may actually increase, but no new image data is added, and you've actually lost data.

Of course, but isn't saving any JPEG to 85 still lossier than saving to 100? 

I think higher file size is perfectly acceptable considering anything lower than 100 is always lossier than 100. 

If somebody is using JPEGView to save an image or convert an image, they only care about either

1) Keeping as much quality as possible when saving an edited image. 
2) Having a JPEG file instead of say a WEBP

I'd go with 100 every time, because Jpegview is more about viewing than editing. I don't use an image viewer to try and manipulate images to that extent. I just care about viewing them and making sure they maintain optimal quality if I do something like rotate them. 

If somebody wants to compress their image, they'd use something else, like https://tinyjpg.com/ or an image editor.

**Comment by Pentaphon** [comment]
>That's inefficient and misleading.

What's inefficient and misleading is to think the average person would use Jpegview to:

1) Download an image from google photos to rotate it
2) Do so in Jpegview, a local image viewer.

When they can rotate an image easily using Google Photos itself. That's what the overwhelming majority of people do. 

https://support.google.com/photos/answer/6128850?hl=en&co=GENIE.Platform%3DAndroid

I daresay that somebody who knows about Jpegview isn't going to use something like Google Photos. 

Do you also think they are going to upload the rotated image back to Google Photos? Probably not.

**Comment by pureby** [comment]
@DeanIsMe The example you provide is not the best one, because 90°/mirroring rotations are lossless if done right (since it's just re-arranging pixel data, not actually re-writing it). JPEGview can do this.

Having said that, I do agree with your statement that "making a very small edit does not warrant 400% filesize increase" ;-)

@Pentaphon IMHO, it's _inefficient and misleading_ to speak on behalf of "average users" and "overwhelming majorities" based solely on your own ideas and personal preference.

**Comment by Pentaphon** [comment]
> it's inefficient and misleading to speak on behalf of "average users" and "overwhelming majorities" based solely on your own ideas and personal preference.

It's not my ideas or personal preference. People who use corporate cloud services like Google photos generally aren't going to stray outside of those apps to use smaller, lesser known apps like JPEGview, especially to do simple operations that already exist in those cloud apps. That's just the way it is.

**Comment by pureby** [comment]
@Pentaphon The logical fallacy within this statement is the same of a kind: "All cats are animals, therefore all animals are cats".

**Comment by pureby** [comment]
> It's not my ideas or personal preference. People who use corporate cloud services like Google photos generally aren't going to stray outside of those apps to use smaller, lesser known apps like JPEGview, especially to do simple operations that already exist in those cloud apps. That's just the way it is.

The logical fallacy within this statement is the same of a kind: "All cats are animals, therefore all animals are cats".

**Comment by sylikc** [comment]
Ok, this discussion has gotten out of hand.  I'm locking it because there's nothing productive that comes out of flaming each other.

There's **a lot of users of JPEGView**, from power users like us, to average Joe's.  I know this first hand based on the varied requests and issues.  I've also seen forums and reddit threads that recommend this viewer just for people looking for a quick viewer to replace the built-in Windows one.  I'm also recommending to the average Joe's non-power users in my life to get feedback what a regular user thinks of it.

Thus, it's really not fair for any of us to talk on behalf of "what other users" want.  It's safest to stick to just what *I* want out of JPEGView, and I don't speak on behalf of anyone either.

_I'm open to having discussions on defaults that target SPECIFIC use cases, but just know that it's likely only a fraction of the **possible** use cases of JPEGView._

Thanks for all your inputs.  This is already now reflected in the pinned issue #140.
